### DISCUSSION TITLE
Keycloak using CAS extension (Missing Parameters : Service)

### DISCUSSION THREAD

**Discussion by PaulGitProject** [keycloak/keycloak#12049] [Q&A]
Hello,

I've tried to use Keycloak OpenJDK with CAS Login Protocol extension from /jacekkow/keycloak-protocol-cas, but I have this page when I try to connect on http://localhost:8080/realms/myrealm/protocol/cas/login :
![3b5b1eb9ad54fbefc3a68d67bbab3268fc773e5f](https://user-images.githubusercontent.com/29106016/168808251-07fe6059-55db-42b5-99f0-0ae5aaa29ada.png)

An whey I go on http://localhost:8080/realms/myrealm/protocol/cas/serviceValidate I have this :
<img width="1088" alt="Capture d’écran 2022-05-17 à 14 11 49" src="https://user-images.githubusercontent.com/29106016/168808198-38d12fe7-9a3e-40c0-8157-c2b8a12f1c65.png">

Here a my settings for Keycloak admin interface :
- First the Realm : 
<img width="902" alt="Capture d’écran 2022-05-17 à 11 25 16" src="https://user-images.githubusercontent.com/29106016/168808327-a6bb084c-79aa-480a-a637-060f624f17be.png">

- Then the client :
<img width="849" alt="Capture d’écran 2022-05-17 à 11 36 03" src="https://user-images.githubusercontent.com/29106016/168808386-88e34d6f-085e-4b95-bdab-288f13691ffb.png">
<img width="822" alt="Capture d’écran 2022-05-17 à 11 29 43" src="https://user-images.githubusercontent.com/29106016/168808409-4f28fd93-9970-414f-9d0a-532a9a08af55.png">

Do you know what am I doing wrong ? Thanks in advance :)

**Comment by RaveNN-0** [comment]
did you manage to find a fix ?
